main/src/cgeo/geocaching/settings/SettingsActivity.java

```diff
@@ -861,6 +861,10 @@ public boolean onPreferenceChange(final Preference preference, final Object valu
             final ListPreference listPreference = (ListPreference) preference;
             final int index = listPreference.findIndexOfValue(stringValue);
 
+            if (isPreference(preference, R.string.pref_selected_language)) {
```

Comment:
good catch, thanks!
Those lines had been there some time along the way, but got lost before I finalized the PR, due to switching between different implementation variants...